runtime/wasmhost/wasmhost.go

```diff
@@ -146,25 +170,11 @@ func (host *wasmHost) GetModuleInstance(ctx context.Context, plugin *plugins.Plu
 		}
 	}
 
-	// Instantiate the plugin as a module.
-	// NOTE: This will also invoke the plugin's `_start` function,
-	// which will call any top-level code in the plugin.
-	mod, err := host.runtime.InstantiateModule(ctx, plugin.Module, cfg)
-	if err != nil {
-		return nil, fmt.Errorf("failed to instantiate the plugin module: %w", err)
-	}
-
-	return mod, nil
-}
-
-func (host *wasmHost) CompileModule(ctx context.Context, bytes []byte) (wazero.CompiledModule, error) {
-	span, ctx := utils.NewSentrySpanForCurrentFunc(ctx)
-	defer span.Finish()
-
-	cm, err := host.runtime.CompileModule(ctx, bytes)
-	if err != nil {
-		return nil, fmt.Errorf("failed to compile the plugin: %w", err)
+	if useStartFuncs {
+		cfg = cfg.WithStartFunctions("_initialize", "_start")
+	} else {
+		cfg = cfg.WithStartFunctions()
```

Comment:
q: what are some cases where we don't want to call the start functions?

Reply:
Good catch. This was left over from an experiment where I thought that on restoring a suspended module we could persist the underlying wasm memory directly.  Turns out Wazero doesn't support that yet, so I unwound that path but left the configuration refactoring in its own function because it seemed appropriate.  But the `useStartFuncs` parameter can be removed.